// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package third_party.pins_infra.thinkit;

option cc_generic_services = false;

// Mode types for the SUT interfaces are connected..
enum InterfaceMode {
  UNKNOWN_MODE = 0;
  DISCONNECTED = 1;
  LOOPBACK = 2;
  CONTROL_INTERFACE = 3;
  TRAFFIC_GENERATOR = 4;
}

// This message represents the mode of interface(s) and the number of
// which are required.
message InterfaceRequirement {
  InterfaceMode interface_mode = 1;
  int32 count = 2;
}

// This message represents interface(s) requirement for a test.
message TestRequirements {
  repeated InterfaceRequirement interface_requirements = 1;
}
